tq/basic_upload.go

```diff
@@ -150,19 +150,23 @@ func (a *basicUploadAdapter) DoTransfer(ctx interface{}, t *Transfer, cb Progres
 func (a *adapterBase) setContentTypeFor(req *http.Request, r io.ReadSeeker) error {
 	uc := config.NewURLConfig(a.apiClient.GitEnv())
 	disabled := !uc.Bool("lfs", req.URL.String(), "contenttype", true)
-	if len(req.Header.Get("Content-Type")) != 0 || disabled {
+	if len(req.Header.Get("Content-Type")) != 0 {
 		return nil
 	}
 
-	buffer := make([]byte, 512)
-	n, err := r.Read(buffer)
-	if err != nil && err != io.EOF {
-		return errors.Wrap(err, "content type detect")
-	}
+	var contentType string
+
+	if !disabled {
+		buffer := make([]byte, 512)
+		n, err := r.Read(buffer)
+		if err != nil && err != io.EOF {
+			return errors.Wrap(err, "content type detect")
+		}
 
-	contentType := http.DetectContentType(buffer[:n])
-	if _, err := r.Seek(0, 0); err != nil {
-		return errors.Wrap(err, "content type rewind")
+		contentType = http.DetectContentType(buffer[:n])
+		if _, err := r.Seek(0, 0); err != nil {
```

Comment:
Nit; we should be using [constants in the `io` package for the seek whence](https://golang.org/pkg/io/#pkg-constants) instead of the literal `0` as the second formal parameter to this function call. For example, I recommend:

```go
if _, err := r.Seek(0, io.SeekStart); err != nil {
        // ...
}
```

I suggest this only because your diff is present around this area, even though you haven't changed this line specifically. If you would like to adopt this change, perhaps you could send it as a preparatory clean-up commit?

Reply:
I've put in a preparatory commit for that. Thanks for the suggestion.